salt/_states/metalk8s_kubernetes.py

```diff
@@ -1505,3 +1505,70 @@ def customresourcedefinition_present(
             'new': res}
 
     return ret
+
+
+def _node_unschedulable(name, value, **kwargs):
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    action = 'cordon' if value else 'uncordon'
+
+    unschedulable = __salt__[
+        'metalk8s_kubernetes.node_unschedulable'](name, **kwargs)
+
+    if unschedulable is None:
+        ret['comment'] = 'Unable to get node {0}'.format(name)
+        return ret
+
+    if unschedulable is value:
```

Comment:
Use `==`, not `is`. Yes, `True` and `False` are singletons, but that shouldn't be relied on.

Reply:
+1

In Python 3, it's kinda safe because True and False are keywords and can't be changed:

```python
>>> True = 0
  File "<stdin>", line 1
SyntaxError: can't assign to keyword
```

But in Python 2 that's not the case and funny things are possible and you can't rely on it:

```python
>>> True, False = False, True # now, let's the fun begin
>>> False
True
>>> True
False
>>> id(True)
140037444246128
>>> True = 42
>>> id(True)
94042690003080
```

But anyway, if you expect boolean it's even more idiomatic to simply write `if unschedulable` or `if not unschedulable` rather than use `==` or `is`.

Reply:
In this case it's an actual comparison between two values (that happen to be Booleans) :)

Reply:
Indeed, I edited my comment after seeing the context ;)

Reply:
Hmm ok just use `is` to match only boolean but I don't care let's use `==`